java/main/com/google/privacy/differentialprivacy/BoundedSum.java

```diff
@@ -124,10 +125,38 @@ public double computeResult() {
 
     resultReturned = true;
     double lInfSensitivity =
-        getLInfSensitivity(params.lower(), params.upper(), params.maxContributionsPerPartition());
-    return params
-        .noise()
-        .addNoise(sum, getL0Sensitivity(), lInfSensitivity, params.epsilon(), params.delta());
+            getLInfSensitivity(params.lower(), params.upper(), params.maxContributionsPerPartition());
+    noisedSum =
+            params
+                  .noise()
+                  .addNoise(sum, getL0Sensitivity(), lInfSensitivity, params.epsilon(), params.delta());
+    return noisedSum;
+  }
+
+  /** ComputeConfidenceInterval computes a confidence interval that contains the true {@link BoundedSum} with
+   *  a probability greater or equal to 1 - alpha using the noised {@link BoundedSum} computed by {@code computeResult()}.
+   */
+  public ConfidenceInterval computeConfidenceInterval(double alpha) {
+    if (!resultReturned) {
+      throw new IllegalStateException("computeResult must be called before calling computeConfidenceInterval.");
+    }
+    ConfidenceInterval confInt =
+            params
+                    .noise()
+                    .computeConfidenceInterval(
+                            noisedSum,
+                            params.maxPartitionsContributed(),
+                            (double) params.maxContributionsPerPartition(),
+                            params.epsilon(),
+                            params.delta(),
+                            alpha);
+    if (params.lower() >= 0) {
```

Comment:
can you please add tests for both ifs?

Like for example, lower >= 0 and you should do clamping
or if lower <. 0 and negative bounds you don't do clamping

the same for upper

Consider not only if condition (min and max are also basically if-condition, so consider all cases)